pages/en/lb3/readmes/loopback-connector-ibmi.md

```diff
@@ -0,0 +1,136 @@
+
+## Major differences from loopback-connector-db2iseries
+- `loopback-connector-db2iseries` is powered by the [IBM Data Server Driver for ODBC and CLI](https://www.ibm.com/support/knowledgecenter/en/SSEPGG_9.7.0/com.ibm.swg.im.dbclient.install.doc/doc/c0023452.html). As noted, this package is powered by the IBM i Access ODBC driver. There may be differences in functionality as a result. From a server protocol perspective, `loopback-connector-db2iseries` module uses [DRDA](https://www.ibm.com/support/knowledgecenter/en/SSGU8G_11.70.0/com.ibm.admin.doc/ids_admin_0206.htm), and this module uses the [IBM i Host Servers](https://www.ibm.com/support/knowledgecenter/en/ssw_ibm_i_74/rzajr/rzajrmst27.htm) used by the rest of the IBM i Access family.
+- `loopback-connector-db2iseries` requires separate licensing to be fully functional. This module does not.
+- This package requires manual installation of prerequisites (see below), whereas `loopback-connector-db2iseries` can handle this automatically as part of the npm install
+
+## Prerequisites
+Before installing this package, you will need an ODBC driver and a driver manager (with development libraries). 
+This package is primarily developed and tested with the [IBM i Access ODBC driver](https://www.ibm.com/support/knowledgecenter/ssw_ibm_i_74/rzaik/rzaikappodbc.htm), which is supported as part of IBM i software maintenance agreements (SWMA) and comes with no additional licensing fees. 
+
+### On IBM i
+- Install the `unixODBC-devel` package. See [the RPM and yum documentation for IBM i](http://ibm.biz/ibmi-rpms) for more detailed steps.
+### On Linux
+- Install the `unixODBC-devel` package with your operating system's package manager (apt-get, zypper, yum, etc).
+- Install the "Linux Application Package" of IBM i Access Client Solutions. Consult [this document](http://www-01.ibm.com/support/docview.wss?uid=nas8N1010355) for assistance.
+### On Windows
+- Install the "Windows Application Package" of IBM i Access Client Solutions. Consult [this document](http://www-01.ibm.com/support/docview.wss?uid=nas8N1010355) for assistance.
+
+
+## Installation
+Once the prerequisites are satisfied, enter the following in the top-level directory of your LoopBack application and install this package:
+
+```
+$ npm install loopback-connector-ibmi 
+```
+
+## Configuration
+
+In LoopBack, datasources are used to store the information about your database so it can be used by the program. Use the [data source generator](https://loopback.io/doc/en/lb3/Data-source-generator.html) to add  to your application:
+
+```
+lb datasource
+```
+
+The datasource generator will then walk you through the process of setting up a datasource:
+1. `Enter the datasource name:` Any name will do, such as the name of the schema you will use or the name of your system.
+2. `Select connector for <name>:` LoopBack 3 does not have knowledge of `loopback-connector-ibmi`, so just press up on the arrow key once and select `other`.
```

Comment:
Let's add `loopback-connector-ibmi` to the list of known connectors please - send a pull request to update the following file:

https://github.com/strongloop/loopback-workspace/blob/master/available-connectors.json

This will add the new connector to both LB 3.x and LB 4.x CLI tools.

The entry should describe connector settings like `connectionString` too, see e.g. the entry for the DB2 connector here: https://github.com/strongloop/loopback-workspace/blob/d5b8061a6275a4f7a38ff8f61357eacb7a614067/available-connectors.json#L75-L99

```
    "settings": {
      "dsn": {
        "type": "string",
        "description": "Connection String dsn to override other settings (eg: DATABASE=MY_DB;HOSTNAME=MY_HOST;PORT=MY_PORT;PROTOCOL=TCPIP;UID=MY_UID;PWD=MY_PWD))"
      },
      "host": {
        "type": "string"
      },
      "port": {
        "type": "number"
      },
      "user": {
        "type": "string"
      },
      "password": {
        "type": "string",
        "display": "password"
      },
      "database": {
        "type": "string"
      },
      "schema": {
        "type": "string"
      }
    },
```

Reply:
Will do! Should I split that into a separate PR or include in this one?

Reply:
You need to open a new pull request for loopback-workspace repository.